src/migration.js

```diff
@@ -89,14 +89,17 @@ module.exports = class Migration {
    *
    * @param {String} method - Name of the method to be called.
    * @param {*} args - Arguments to be used when called the method.
-   * @returns {*|Promise}
+   * @returns {Promise}
    * @private
    */
-  _exec(method, args) {
-    var migration  = this.migration();
-    var fun        = migration[method];
+  async _exec(method, args) {
+    const migration = await this.migration();
+    let fun = migration[method];
+    if (migration.default) {
+      fun = migration.default[method] || migration[method];
+    }
     if (!fun) return Bluebird.reject('Could not find migration method: ' + method);
-    var wrappedFun = this.options.migrations.wrap(fun);
+    const wrappedFun = this.options.migrations.wrap(fun);
 
     return wrappedFun.apply(migration, args);
```

Comment:
Here you're mixing promise rejections and async/await. If you want to reject inside of an async function, just throw an exception - it translates to the same thing.

Also, if wrappedFun returns a promise (and I believe it does)  you should await that as well, rather than simply returning it.